### ISSUE TITLE
query of CAT balance doesn't work

### ISSUE THREAD

**Issue by Chia-Inventory-The-Green-Whale** [ageorge95/WILLOW-chia-forks-instant-wallet-balance#12]
query of CAT balance doesn't work, i guess it is due to new CAT2 standard.
more, the way you query CAT balance seems working only for transactions related to one single address, so the balance may not be correct.

**Comment by ageorge95** [comment]
Hi,

-> the CAT2 standard will be implemented at some point (I am not really into CATs so that is a really low priority for me)
-> the balance for both CATs and the actual assets should be queried based on what input you provide - if you give it a single address/ a list of addresses, it will query only those addresses; but if you feed the scripts with a mnemonic it will query the first 500 (default) addresses of that mnemonic.

So, long story short, the CAT2 standard will be implemented soon(TM) and the behavior described by you regarding the querying of the balance should not happen.

With your permission I will keep the issue open and rename it so that it reflects the CAT2 implementation feature.

(if you really think that the querying is done wrong give me more details and I will look into it 🙂)
